consensus/consensus-types/src/sync_info.rs

```diff
@@ -191,9 +191,9 @@ impl SyncInfo {
     }
 
     pub fn has_newer_certificates(&self, other: &SyncInfo) -> bool {
-        self.highest_certified_round() > other.highest_certified_round()
+        self.highest_ledger_info_round() > other.highest_ledger_info_round()
```

Comment:
why the order matter?

Reply:
consider a node with a qc of round 200, but it has slow execution at HLI 10, and it was set to be sync_only. So this blocks the sync_up path until the local qc is smaller than remote qc. However, during sync_only, the node can still update the local qc through the process_vote path?

Reply:
worth a comment about the ordering

Reply:
added

Reply:
> consider a node with a qc of round 200, but it has slow execution at HLI 10, and it was set to be sync_only. So this blocks the sync_up path until the local qc is smaller than remote qc. However, during sync_only, the node can still update the local qc through the process_vote path?

actually this is wrong. I was thinking of comparisons...